pkg/admissioncontroller/webhook/admission/authorizationconfig/add.go

```diff
@@ -17,13 +37,87 @@ const (
+func AddToManager(mgr manager.Manager) error {
 	webhook := &admission.Webhook{
-		Handler:      h,
+		Handler: NewHandler(
+			mgr.GetLogger().WithName("webhook").WithName(HandlerName),
+			mgr.GetAPIReader(),
+			mgr.GetClient(),
+			admission.NewDecoder(mgr.GetScheme()),
+		),
 		RecoverPanic: ptr.To(true),
 	}
 
 	mgr.GetWebhookServer().Register(WebhookPath, webhook)
 	return nil
 }
+
+// NewHandler returns a new handler for validating authorization configuration.
+func NewHandler(log logr.Logger, apiReader, c client.Reader, decoder admission.Decoder) admission.Handler {
+	return &configvalidator.Handler{
+		Logger:    log,
+		APIReader: apiReader,
+		Client:    c,
+		Decoder:   decoder,
+
+		ConfigMapKind:    "authorization configuration",
+		ConfigMapDataKey: "config.yaml",
+		GetConfigMapNameFromShoot: func(shoot *gardencore.Shoot) string {
+			return gardencorehelper.GetShootAuthorizationConfigurationConfigMapName(shoot.Spec.Kubernetes.KubeAPIServer)
+		},
+		AdmitConfig: admitConfig,
+	}
+}
+
+var decoder runtime.Decoder
+
+func init() {
+	scheme := runtime.NewScheme()
+	schemeBuilder := runtime.NewSchemeBuilder(apiserverv1beta1.AddToScheme, apiserverv1alpha1.AddToScheme, apiserver.AddToScheme)
+	utilruntime.Must(schemeBuilder.AddToScheme(scheme))
+	decoder = serializer.NewCodecFactory(scheme).UniversalDecoder()
+}
+
+func admitConfig(authorizationConfigurationRaw string, shoots []*gardencore.Shoot) (int32, error) {
+	obj, schemaVersion, err := decoder.Decode([]byte(authorizationConfigurationRaw), nil, nil)
+	if err != nil {
+		return http.StatusUnprocessableEntity, fmt.Errorf("failed to decode the provided authorization configuration: %w", err)
+	}
+
+	authorizationConfig, ok := obj.(*apiserver.AuthorizationConfiguration)
+	if !ok {
+		return http.StatusInternalServerError, fmt.Errorf("failed to cast to authorization configuration type: %v", schemaVersion)
+	}
+
+	var authorizerNames []string
+
+	for i, webhook := range authorizationConfig.Authorizers {
+		if webhook.Type == apiserver.TypeWebhook && webhook.Webhook != nil {
+			authorizerNames = append(authorizerNames, webhook.Name)
+			// We do not allow users to set the connection info in the webhook configurations, so let's first validate this:
+			if !apiequality.Semantic.DeepEqual(webhook.Webhook.ConnectionInfo, apiserver.WebhookConnectionInfo{}) {
+				return http.StatusUnprocessableEntity, fmt.Errorf("provided invalid authorization configuration: %v", field.ErrorList{field.Forbidden(field.NewPath("authorizers").Index(i).Child("connectionInfo"), "connectionInfo is not allowed to be set")})
+			}
+			// At this point, we have ensured that the connection info is not set. However, below validation function
+			// expects it to be set, so let's fake it here to make the function pass (this is not persisted anywhere).
+			authorizationConfig.Authorizers[i].Webhook.ConnectionInfo.Type = apiserver.AuthorizationWebhookConnectionInfoTypeInCluster
+		}
+	}
+
+	if len(shoots) == 1 {
```

Comment:
Isn't it more correct to validate the `authorizerNames` against all of the shoots that reference the `ConfigMap` (even though this wasn't done before)?

I'm imagining the following case  that could happen currently:
1. 2 shoots have enabled structured authorization. 
2. Each of them has 1 entry in `.spec.kubernetes.kubeAPIServer.structuredAuthorization.kubeconfigs[]`. 
3. An operator adds an additional entry to that array for the first shoot. 
4. The operator modifies the referenced structured authorization ConfigMap to add an additional webhook that would match the new kubeconfig entry which is only available in one of the shoots
5. Currently, the validation passes as the length of shoots is > 1

Reply:
Yes, definitely, good catch. Unfortunately, I don't remember the exact reason why it wasn't done this way in the initial implementation, but it was a deliberate decision. I think it didn't fit quite well into the validator structure copied from the other existing validators at this time. Anyways, I added it now - I think it's pretty straight-forward now with the new approach :) PTAL